File: Basic/Max Sum Subarray of size K.java
 
// { Driver Code Starts
//Initial Template for Java
import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0)
        {
            
            int N=sc.nextInt();
            int K=sc.nextInt();
            ArrayList<Integer> Arr = new ArrayList<Integer>(N); 
  
            // Appending new elements at 
            // the end of the list 
            for (int i = 0; i < N; i++) {
                int x = sc.nextInt();
                Arr.add(x); 
            }

            Solution ob = new Solution();
            System.out.println(ob.maximumSumSubarray(K,Arr,N));
        }
    }
}
// } Driver Code Ends


//User function Template for Java
class Solution{
    static int maximumSumSubarray(int K, ArrayList<Integer> Arr,int N){
        // code here
        if (K>N) return -1;
        
        
        int somma = 0;
        int massimo = 0;
        int inizio = 0;
        
        for (int i=0; i<N; i++) {
            somma += Arr.get(i);
                
            if (i>=K) {
                somma -= Arr.get(inizio);
                inizio++;
            }
            massimo = Math.max(massimo, somma);
        }
        return massimo;
    }
}

File: Easy/Max Sum Subarray of size K.py
#User function Template for python3
class Solution:
    def maximumSumSubarray (self, K, Arr, N):
        # code here
        sum  = 0
        
        if (K >= N):
            for it in Arr:
                sum += it
            return sum
        
        for i in range(K):
            sum += Arr[i]
            
        ans = sum
        
        for i in range(K, N):
            sum -= Arr[i-K]
            sum += Arr[i]
            ans = max(ans, sum)
        
        return ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3
if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        
        N,K = input().split()
        N = int(N)
        K = int(K)
        Arr = list( map(int,input().strip().split(" ")))

        ob = Solution()
        print(ob.maximumSumSubarray(K,Arr,N))
# } Driver Code Ends